src/data/calendar.js

```diff
@@ -0,0 +1,36 @@
+
+module.exports = {
+    /**
+    * Puts all cards with due dates into a .ics file
+    * @param {String} boardId The board's id.
+    * @returns A list of list(column) objects.
+    */
+    makeCal: async (boardId) => {
+        if(!boardId || !error_handler.checkObjectId(id)) 
+            throw new Error("Board id is not valid.");
+
+        const boardCollection = await boards();
+        const board = await boardCollection.findOne({_id: ObjectId(boardId)});
+        
+        let cal = ics();
+        for(let x=0; x<board.cards.length; x++) {
+            let card = board.cards[x];
+            cal.addEvent(card.cardName, card.description, board.boardName, dueDate.date, dueDate.date)
+        }
+
+        let ret = cal.download(board.boardName);
+        if(ret === false) {
+            throw new Error("Could not download file.");
+        }
+
+        return ret;
```

Comment:
What specifically is returned here?

Reply:
So this function pops up with a download prompt like typical google or whatever, so if that doesn't fail I just return true so that we know everything worked properly